### DISCUSSION TITLE
Calculate actual scaleX, scaleY, angle of every object when scaling or rotating a mutiple selected objects.

### DISCUSSION THREAD

**Discussion by tabvn** [fabricjs/fabric.js#7351] [Q&A]
Hi.

Calculate actual scaleX, scaleY, angle of every object when scaling or rotating a multiple selected objects..

because when we select multiple objects and rotate or scaling we have event.target._objects, however actual of angle, scaleX, scaleY inside of object is not updating yet.

do we have func built to calculate yet.

Thank you.

**Comment by melchiar** [comment]
The method you're looking for is `calcTransformMatrix();`. You can use it like this:

```
var matrix = event.target._objects[0].calcTransformMatrix();
var options = fabric.util.qrDecompose(matrix);
```

The `options` object will have the transformed values you need. See http://fabricjs.com/using-transformations to better understand the transformMatrix system.

**Reply by tabvn** [reply]
thank you @melchiar for your helps.
